app/src/main/java/me/hufman/androidautoidrive/music/spotify/SpotifyWebApi.kt

```diff
@@ -72,6 +76,79 @@ class SpotifyWebApi private constructor(val context: Context, val appSettings: M
 		spotifyAppControllerCaller = null
 	}
 
+	/**
+	 * Creates a private playlist with the provided name. The newly created playlist's [PlaylistUri]
+	 * is returned.
+	 */
+	suspend fun createPlaylist(playlistName: String): PlaylistUri? {
+		try {
+			return webApi?.playlists?.createClientPlaylist(name = playlistName, description = L.MUSIC_TEMPORARY_PLAYLIST_DESCRIPTION, public = false)?.uri
+		} catch (e: SpotifyException.AuthenticationException) {
+			Log.e(TAG, "Failed to create playlist $playlistName due to authentication error with the message: ${e.message}")
+			authStateManager.addAccessTokenAuthorizationException(e)
+			createNotAuthorizedNotification()
+			webApi = null
+		} catch (e: Exception) {
+			Log.e(TAG, "Exception occurred when trying to create playlist $playlistName with the message: ${e.message}")
+		}
+		return null
+	}
+
+	/**
+	 * Adds the provided list of songs to the specified playlist.
+	 */
+	suspend fun addSongsToPlaylist(playlistId: String, songs: List<MusicMetadata>) {
+		try {
+			webApi?.playlists?.addTracksToClientPlaylist(playlistId, *songs.map { it.mediaId!! }.toTypedArray())
+		} catch (e: SpotifyException.AuthenticationException) {
+			Log.e(TAG, "Failed to add songs to playlist $playlistId due to authentication error with the message: ${e.message}")
+			authStateManager.addAccessTokenAuthorizationException(e)
+			createNotAuthorizedNotification()
+			webApi = null
+		} catch (e: Exception) {
+			Log.e(TAG, "Exception occurred when trying to add songs to playlist $playlistId with the message: ${e.message}")
+		}
+	}
+
+	/**
+	 * Replaces the songs of the specified playlist with the provided list of songs.
+	 */
+	suspend fun replacePlaylistSongs(playlistId: String, songs: List<MusicMetadata>) {
+		try {
+			webApi?.playlists?.replaceClientPlaylistTracks(playlistId, *songs.map { it.mediaId!! }.toTypedArray())
+		} catch (e: SpotifyException.AuthenticationException) {
+			Log.e(TAG, "Failed to replace playlist $playlistId songs due to authentication error with the message: ${e.message}")
+			authStateManager.addAccessTokenAuthorizationException(e)
+			createNotAuthorizedNotification()
+			webApi = null
+		} catch (e: Exception) {
+			Log.e(TAG, "Exception occurred when trying to replace playlist $playlistId songs with the message: ${e.message}")
+		}
+	}
+
+	/**
+	 * Returns the [SpotifyUri] associated to the provided playlist name or null if it is not found.
+	 */
+	suspend fun getPlaylistUri(playlistName: String): SpotifyUri? {
+		try {
+			val playlists = webApi?.playlists?.getClientPlaylists(15,0)
```

Comment:
Providing a limit of 15 seems too small, I'd be fine setting this to the max of 50, and since it's an async function, it might be feasible to implement pagination easily enough.

Reply:
Agreed, updated the limit to 50 and also added pagination support for the getClientPlaylists built in pagination system